app/helpers/elasticsearch_helper.rb

```diff
@@ -32,11 +33,31 @@ def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys)
         }
       end
     end
-    matching_taxa.values
+
+    if project_id.present?
+      filter_by_project(matching_taxa.values, project_id)
+    else
+      matching_taxa.values
+    end
   end
 
   private
 
+  # Took 250ms in local testing on real data
```

Comment:
Local testing taxon_counts table will be orders of magnitude smaller than prod though, just FYI... still needs some testing on actual prod size data. Even staging only has a small subset of prod's taxon_counts

Reply:
If it holds up in prod, amazing!

Reply:
ok sorry I didn't realize.... I'll test this query in the rails console in prod.

Reply:
The explain results look good. 

```=> EXPLAIN for: SELECT DISTINCT `taxon_counts`.* FROM `taxon_counts` INNER JOIN `pipeline_runs` ON `pipeline_runs`.`id` = `taxon_counts`.`pipeline_run_id` INNER JOIN `samples` ON `samples`.`id` = `pipeline_runs`.`sample_id` INNER JOIN `projects` ON `projects`.`id` = `samples`.`project_id` WHERE `samples`.`project_id` = 70 AND `taxon_counts`.`tax_id` IN (59512, 11040, 1649490, 182402, 28067) AND (tax_id > 0) AND `taxon_counts`.`count_type` IN ('NT', 'NR') AND (count > 5)
+----+-------------+---------------+-------+------------------------------------------+----------------------------------+---------+-----------------------------+------+------------------------------------+
| id | select_type | table         | type  | possible_keys                            | key                              | key_len | ref                         | rows | Extra                              |
+----+-------------+---------------+-------+------------------------------------------+----------------------------------+---------+-----------------------------+------+------------------------------------+
|  1 | SIMPLE      | projects      | const | PRIMARY                                  | PRIMARY                          | 8       | const                       |    1 | Using index; Using temporary       |
|  1 | SIMPLE      | samples       | ref   | PRIMARY,index_samples_name_project_id    | index_samples_name_project_id    | 9       | const                       |   15 | Using index                        |
|  1 | SIMPLE      | pipeline_runs | ref   | PRIMARY,index_pipeline_runs_on_sample_id | index_pipeline_runs_on_sample_id | 9       | idseq_prod.samples.id       |    1 | Using index                        |
|  1 | SIMPLE      | taxon_counts  | ref   | index_pr_tax_hit_level_tc                | index_pr_tax_hit_level_tc        | 9       | idseq_prod.pipeline_runs.id |  216 | Using index condition; Using where |
+----+-------------+---------------+-------+------------------------------------------+----------------------------------+---------+-----------------------------+------+------------------------------------+
```

Reply:
yay!